### DISCUSSION TITLE
<Label> rendered as a <span role="label">

### DISCUSSION THREAD

**Discussion by xavipolo** [radix-ui/primitives#1742] [Help] [answered]
Hi, 

I am using the Label primitive like in

```
    <LabelPrimitive.Root
      htmlFor={htmlFor}
      className={
        className ? className : "m-2 bg-gray-200 text-sm dark:bg-gray-600"
      }
    >
      {children}
    </LabelPrimitive.Root>
```

but looking at the generated HTML I see that it is rendered as:

`<span role="label" id="radix-:R4u6:" class="m-2 bg-gray-200 text-sm dark:bg-gray-600">This is the label</span>`

instead of an html `<label ... >`.

Does anyone know what could make it behave like this?  The problem is that then I don't get the htmlFor.

Thanks.

**Comment by joaom00** [comment, marked as answer]
Hi @xavipolo,

Which version are you using? The native label element is used in version `2.0.0`. You can see the full changelog [here](https://www.radix-ui.com/docs/primitives/overview/releases).

**Reply by xavipolo** [reply]
It was the version, updated.
Thank you!
